### DISCUSSION TITLE
How to make a command line version of https://cdn.tailwindcss.com/?

### DISCUSSION THREAD

**Discussion by jvns** [tailwindlabs/tailwindcss#10614] [Help]
I'd like to make myself a command line version of the Play CDN version of tailwind. 

The idea is:

* one single Javascript file no dependencies (`my_tailwind.js`)
* it just uses Tailwind's default settings (no customization)
* it reads HTML from stdin: `cat file1.html  file2.html | node my_tailwind.js`
* it outputs all of the CSS classes used in the HTML it read from stdin
* I can commit the Javascript file to my Git repo so that it don't have any external dependencies except node

Is this something that I could easily build myself? I'm not super familiar with the internals of Tailwind so I'd love any pointers. I realize the normal way to do this is with the standalone CLI, but I'm curious about whether this is possible.

**Comment by tlgimenes** [comment]
Hi! I'm also curious for this one. I'm building some websites on top of deno and I would love a function that:
```
import generator from 'tailwindcss/some/path.ts'

const css = generator(tailwindConfig, tokens)
```

where `tokens` is an array of used tailwind classes and tailwindConfig is my resolved tailwind config. I really don't understand why we need these complex postCSS setups

**Comment by sntran** [comment]
Looks like Tailwind is moving away from PostCSS so it'd get easier to have a simpler pipeline.

In the mean time, you may be able to achieve what you want with `twind`, especially when you just want default settings.

**Reply by tlgimenes** [reply]
Hi @sntran! Nice to hear tailwind is moving away from PostCSS, do you have any links to back this up? I'm really interested on reading more on this topic.

Regarding `twind`, I tried it this year and even tried to [contribute](https://github.com/tw-in-js/twind/pull/467), however this project seems a little bit dead and really harms performance. I'm really looking into a CSS-only solution backed up by a JIT compiler. `twind` would be perfect if not for the css-in-js that REALLY slows down my websites. 

I wrote some [thoughts on CSS](https://github.com/denoland/fresh/discussions/1120) if you want to share your thoughts. Thanks!

**Reply by sntran** [reply]
Adam has a couple of tweets about this, for example: https://twitter.com/adamwathan/status/1649391725272604672

Agreed with you about `twind`. While it's nice not having a build step and can use it in different environments, not being able to reuse existing Tailwind config file is a bummer.

Good to see a fellow Deno user!

**Reply by tlgimenes** [reply]
Thanks for the links @sntran! I didn't know about this lightning CSS! really looking forward to integrating to Deno Fresh!
